packages/block-editor/src/components/block-list/content.scss

```diff
@@ -85,7 +85,6 @@
 	.block-editor-block-list__block.is-highlighted,
 	.block-editor-block-list__block.is-highlighted ~ .is-multi-selected,
 	&.is-navigate-mode .block-editor-block-list__block.is-selected,
-	& .is-block-moving-mode.block-editor-block-list__block.has-child-selected,
```

Comment:
The `.is-block-moving-mode` class is given _only_ to the selected block. Therefore, there should never be `.is-block-moving-mode` class and `.has-child-selected` class at the same time.